### ISSUE TITLE
[WAGON-554] Tests with checkin rely on global Git config

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-wagon#627]
**[Matthieu Brouillard](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mcfoggy)** opened **[WAGON-554](https://issues.apache.org/jira/browse/WAGON-554?redirect=false)** and commented

Inside maven-wagon-scm project, the several tests in `org.apache.maven.wagon.providers.scm.ScmGitExeWagonTest` expect a global user.name and user.email to be set.

On installations where .gitconfig contains the following:

```
[user]
useConfigOnly = true
```

The tests will fail.

It would be better after repository creation to configure a user name & email at project level.

---

**Affects:** 3.3.1

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Sample outout of ASF Jenkins:

```
 linux-jdk11 / Build linux-jdk11 / org.apache.maven.wagon.providers.scm.ScmGitExeWagonTest.testWagonPutDirectory
 Fehlerdetails

Error interacting with SCM: Unable to commit file. The git-commit command failed. 
*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal:********@asf929.gq1.ygridcore.net>) not allowed

 Stacktrace

org.apache.maven.wagon.TransferFailedException: 
Error interacting with SCM: Unable to commit file. The git-commit command failed. 
*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal:********@asf929.gq1.ygridcore.net>) not allowed

Caused by: org.apache.maven.scm.ScmException: 
Unable to commit file. The git-commit command failed. 
*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal:********@asf929.gq1.ygridcore.net>) not allowed

```

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

We can borrow the code from Maven SCM.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [38e5b3237dd3e94187262bf9bf6fc2f70c6678bc](https://gitbox.apache.org/repos/asf?p=maven-wagon.git;a=commit;h=38e5b3237dd3e94187262bf9bf6fc2f70c6678bc).
